hs_core/views/resource_metadata_rest_api.py

```diff
@@ -0,0 +1,243 @@
+    type = serializers.CharField(required=False)
+    value = serializers.CharField(required=False)
+
+    class Meta:
+        model = Relation
+
+
+class CoreMetaDataSerializer(serializers.Serializer):
+    title = serializers.CharField(required=False)
+    creators = CreatorSerializer(required=False, many=True)
+    contributors = PartySerializer(required=False, many=True)
+    coverages = CoverageSerializer(required=False, many=True)
+    dates = DateSerializer(required=False, many=True)
+    description = serializers.CharField(required=False)
+    formats = FormatSerializer(required=False, many=True)
+    funding_agencies = FundingAgencySerializer(required=False, many=True)
+    identifiers = IdentifierSerializer(required=False, many=True)
+    language = serializers.CharField(required=False)
+    rights = serializers.CharField(required=False)
+    type = serializers.CharField(required=False)
+    publisher = serializers.CharField(required=False)
+    sources = SourceSerializer(required=False, many=True)
+    subjects = SubjectSerializer(required=False, many=True)
+    relations = RelationSerializer(required=False, many=True)
+
+    class Meta:
+        model = CoreMetaData
+
+
+class MetadataElementsRetrieveUpdate(generics.RetrieveUpdateDestroyAPIView):
+    """
+    Retrieve resource science (Dublin Core) metadata
+
+    REST URL: /hsapi/resource/{pk}/scimeta/elements/
+    HTTP method: GET
+
+    :type pk: str
+    :param pk: id of the resource
+    :return: resource science metadata as JSON document
+    :rtype: str
+    :raises:
+    NotFound: return json format: {'detail': 'No resource was found for resource id:pk'}
+    PermissionDenied: return json format: {'detail': 'You do not have permission to perform
+    this action.'}
+
+    REST URL: /hsapi/resource/{pk}/scimeta/elements/
+    HTTP method: PUT
+
+    :type pk: str
+    :param pk: id of the resource
+    :type request: JSON formatted string
+    :param request: resource metadata
+    :return: updated resource science metadata as JSON document
+    :rtype: str
+    :raises:
+    NotFound: return json format: {'detail': 'No resource was found for resource id':pk}
+    PermissionDenied: return json format: {'detail': 'You do not have permission to perform
+    this action.'}
+    ValidationError: return json format: {parameter-1': ['error message-1'],
+    'parameter-2': ['error message-2'], .. }
+    """
+    ACCEPT_FORMATS = ('application/json',)
+
+    allowed_methods = ('GET', 'PUT')
+
+    serializer_class = CoreMetaDataSerializer
+
+    def get(self, request, pk):
+        view_utils.authorize(request, pk, needed_permission=ACTION_TO_AUTHORIZE.VIEW_METADATA)
+        resource = hydroshare.get_resource_by_shortkey(shortkey=pk)
+        serializer = CoreMetaDataSerializer(resource.metadata)
+        return Response(data=serializer.data, status=status.HTTP_200_OK)
+
+    def put(self, request, pk):
+        # Update science metadata based on resourcemetadata.xml uploaded
```

Comment:
This comment seems not right. We are not updating resource metadata from an uploaded xml file.